Remove space and background color on menubar

tuaris opened the pull request:
Removes extra space and background color from bottom of menu bar so it looks less out of place.

## Screenshots

[Before](http://venus.morante.net/downloads/unibia/screenshots/firefox-tabs-in-middle-before.png):
![image](https://user-images.githubusercontent.com/5660265/120900111-11277e80-c601-11eb-8cf0-a808f3dee0d2.png)

[After](http://venus.morante.net/downloads/unibia/screenshots/firefox-tabs-in-middle-after.png): 
![image](https://user-images.githubusercontent.com/5660265/120900117-15539c00-c601-11eb-96fa-e2df121b9df4.png)

MrOtherGuy commented:
That's not so simple sadly.

For one, the menubar height won't actually collapse to 20px *unless* one enables native titlebar. And, if one places buttons/bookmarks anything else into menubar, then 20px certainly is not tall enough.

Basically, that 20px only looks good if native titlebar is enabled AND the menubar doesn't contain any other elements except menubar items. And that is not an assumption I would do.

However, it might be possible to make the situation a bit better. I applied a patch in 6f2d9f5 that I think should have your desired behavior by default, if you keep native titlebar enabled. 

As for the color... I don't want to change appearance of things unless it's the purpose of the stylesheet. So changing the background color of main menubar doesn't really fit here. But feel free to use it yourself, and yes I think it does look better that way.

Thank you for the idea!

tuaris commented:
Ah yes, I totally forgot that I have the native title bar option enabled and that it even existed.  I removed my changes and tested your new [tabs_on_bottom_menubar_on_top_patch.css](https://github.com/MrOtherGuy/firefox-csshacks/blob/6f2d9f55937f7151f7520b33ab8442c817d6e014/chrome/tabs_on_bottom_menubar_on_top_patch.css) file (I moved my background color changes to userChrome.css). 

It works well with the exception of one case.  When the browser has the native title bar disabled and it's maximized, the menu bar gets moved too far up, cutting off the text:

![image](https://user-images.githubusercontent.com/5660265/120903490-5b662b00-c614-11eb-85a6-01caca0f3767.png)

Here it is with the window un-maximized:
![image](https://user-images.githubusercontent.com/5660265/120903527-92d4d780-c614-11eb-9343-c7327131a11a.png)

Strangely, when you have the toolbar customize tab open, this goes away:
![image](https://user-images.githubusercontent.com/5660265/120903504-6faa2800-c614-11eb-863b-d673e0d97c5e.png)

MrOtherGuy commented:
That would be caused by tabs_on_bottom.css not applying the correct value for Win7. It only sets --uc-titlebar-padding to 8px on Win10, the rest falls back to 0px. I have no way to test Win7 so I don't know what I could have set it to.

But since you're here, would you please set the value at [tabs_on_bottom.css Line 18](https://github.com/MrOtherGuy/firefox-csshacks/blob/master/chrome/tabs_on_bottom.css#L18) to something to test what value works. Actually the 8px that's used on Windows10 looks pretty close, but maybe a bit less is better. Optimally it should be just so that when in maximize mode, the cursor hits the menubar buttons when it is at the top of the screen. 

So if you would, test what value works, get back to me and I'll patch the file to add better support for Win7. This would actually be useful for couple of other styles as well, although I don't know how much use Win7 sees these days. But nonetheless, it would be an improvement.

Thanks in advance!

tuaris commented:
I moved over to a clean environment to make sure I didn't have anything else contaminating the settings.  Currently using only `tabs_on_bottom.css` and the new `tabs_on_bottom_menubar_on_top_patch.css` file.  Native title bar is disabled, menu bar is enabled.

Tweaking line 18 in `tabs_on_bottom.css` between 8px and 10px seems to be a sweet spot.  Here's what it looks like while maximized and set to 8px:
![image](https://user-images.githubusercontent.com/5660265/120905860-07af0e00-c623-11eb-8227-cabc1a17707b.png)

Same, but un-maximized window:
![image](https://user-images.githubusercontent.com/5660265/120905888-2dd4ae00-c623-11eb-8dd7-9d8ce7f423c8.png)

While customize toolbar mode is active:
![image](https://user-images.githubusercontent.com/5660265/120905906-4b097c80-c623-11eb-9320-7564d87bcc8d.png)

It seems to line up better when in customize toolbar mode.

Here's something I noticed when you have both the menu bar and native toolbar disabled while using only `tabs_on_bottom.css` with or without the tweak on line 18.

![image](https://user-images.githubusercontent.com/5660265/120905972-c10de380-c623-11eb-9430-de494f4a3e82.png)

Same when maximized:
![image](https://user-images.githubusercontent.com/5660265/120905993-e7cc1a00-c623-11eb-9307-1a168ff7c499.png)

Seems like in Windows 7 you have to enable the native title bar when using `tabs_on_bottom.css` regardless of the menu bar setting.

tuaris commented:
Well, seems like Windows 10 has the same issue while using only `tabs_on_bottom.css`.
![image](https://user-images.githubusercontent.com/5660265/120906454-742c0c00-c627-11eb-8be7-8a0c185810b3.png)

Must be because of the new Proton theme.

MrOtherGuy commented:
Thanks for testing. I updated tabs_on_bottom.css and other files that do that sort of thing in #86f8cf4 so now the same extra padding applies both on Win7 and Win10. 

That window control overlap isn't a problem. tabs_on_bottom.css just requires [window_control_placeholder_support.css](https://github.com/MrOtherGuy/firefox-csshacks/blob/master/chrome/window_control_placeholder_support.css) if native titlebar is disabled. That style is like a shared resource that multiple styles use with a purpose to create a placeholder space in nav-bar where window controls can fit.
